core/domain/collection_services.py

```diff
@@ -134,23 +135,35 @@ def get_collection_from_model(collection_model, run_conversion=True):
     """
 
     # Ensure the original collection model does not get altered.
-    versioned_collection = {
+    versioned_collection_contents = {
```

Comment:
Maybe we could just do:

```python
versioned_collection_contents = copy.deepcopy(collection_model.collection_contents)
versioned_collection_contents['schema_version'] = collection_model.schema_version
```

It's a bit awkward to have ``versioned_collection_contents['collection_contents']``, though I'm a bit ambivalent and am okay with it staying as is (since it has parity with explorations). It just seems highly unlikely we would ever have a key conflict in the contents dict with ``schema_version`` and have an opportunity to clean this up, whereas with exploration states it's more complicated since key names are user provided.